tests/e2e/configurer/configurer.go

```diff
@@ -0,0 +1,513 @@
+package configurer
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path"
+	"path/filepath"
+	"testing"
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/ory/dockertest/v3"
+	"github.com/ory/dockertest/v3/docker"
+	"github.com/stretchr/testify/require"
+	rpchttp "github.com/tendermint/tendermint/rpc/client/http"
+
+	"github.com/osmosis-labs/osmosis/v7/tests/e2e/chain"
+	dockerImages "github.com/osmosis-labs/osmosis/v7/tests/e2e/configurer/docker"
+	"github.com/osmosis-labs/osmosis/v7/tests/e2e/util"
+)
+
+type Configurer interface {
+	ConfigureChains() error
+
+	ClearResources() error
+
+	GetChainConfig(chainIndex int) ChainConfig
+
+	RunSetup() error
+
+	RunValidators() error
+
+	RunIBC() error
+
+	SendIBC(srcChain *chain.Chain, dstChain *chain.Chain, recipient string, token sdk.Coin)
+
+	CreatePool(chainId string, valIdx int, poolFile string)
+}
+
+type BaseConfigurer struct {
+	chainConfigs   []*ChainConfig
+	dockerImages   *dockerImages.ImageConfig
+	dockerPool     *dockertest.Pool
+	dockerNetwork  *dockertest.Network
+	valResources   map[string][]*dockertest.Resource
+	hermesResource *dockertest.Resource
+	setupTests     setupFn
+	t              *testing.T
```

Comment:
I do like that this is moving the configuration logic to being not in `_test` logic, but instead be importable!